### DISCUSSION TITLE
Does `ignoreDeps` affect lockfile maintenance?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#17982] [Request Help] [answered]
## Does `ignoreDeps` affect lockfile maintainance?

We're using the `ignoreDeps` config option so that Renovate ignores some `msw` dependencies. When Renovate performs lockfile maintainance, it still touches the "ignored dependencies":

- https://github.com/octoclairvoyant/octoclairvoyant-webapp/pull/1164#discussion_r979886602

The [Renovate docs for `ignoreDeps`](https://docs.renovatebot.com/configuration-options/#ignoredeps) don't explain  if Renovate supports _ignoring dependencies in lockfiles_. I guess that we just don't support ignoring dependencies in a lockfile?

Maybe what we really need is to to tell `pnpm` itself what we want, maybe with a "resolution" or something like that?

### Full config

```json
{
	"extends": [
		"config:base",
		"helpers:pinGitHubActionDigests",
		":automergeTypes",
		"npm:unpublishSafe"
	],
	"prCreation": "not-pending",
	"lockFileMaintenance": {
		"enabled": true,
		"automerge": true
	},
	"internalChecksFilter": "strict",
	"ignoreDeps": ["msw", "headers-polyfill", "@mswjs/interceptors"],
	"packageRules": [
		{
			"description": [
				"Group all non-major pnpm development dependencies together, get updates once a month"
			],
			"matchManagers": ["npm"],
			"matchDepTypes": ["devDependencies"],
			"matchUpdateTypes": ["minor", "patch"],
			"groupName": "devDependencies (non-major)",
			"extends": ["schedule:monthly"]
		},
		{
			"description": "Get pnpm updates once a month",
			"matchDepTypes": ["packageManager"],
			"matchPackageNames": ["pnpm"],
			"extends": ["schedule:monthly"]
		},
		{
			"description": "Automerge ESLint and Prettier updates",
			"matchDepTypes": ["devDependencies"],
			"matchPackagePatterns": ["eslint", "prettier"],
			"automerge": true
		}
	]
}
```

**Comment by rarkins** [comment, marked as answer]
It does not affect. Renovate just calls "npm install" to let it install whatever it decides

**Reply by rarkins** [reply]
Sounds like you should pin the dependencies you don't want up to update, if they use ranges currently. If they are transitive dependencies then you may be able to use "overrides"

**Reply by Belco90** [reply]
Thanks! I think that's actually the best way to fix transitive dependencies. I'll give it a try with pnpm.
